### PR TITLE
Insertions out of metadata

### LINKED ISSUE #372: Move insertions out of metadata
* There should not be a metadata type for insertions
* In the tsv/fasta preprocessing format: Currently, insertions are provided to SILO in the metadata.tsv file. Think of how else insertions should be provided
* In NDJSON format, insertions can be provided separately from metadata already.
* This solves a couple of issues:
  * currently it's possible to aggregated by insertions
  * https://github.com/GenSpectrum/LAPIS-SILO/pull/354
  * "Workarounds" in the code, where you need to consider metadata types that are not insertions (which is the case quite often)

### PR DESCRIPTION
**Pull request by Taepper**
resolves #372 

### Summary

Insertions are no metadata, but are specific to an aligned sequence. This is now reflected in the code. The database config does no longer contain the metadata, but instead there is an nuc_insertions and aa_insertions required, which give all insertions for nucleotide sequences and amino acid sequences respectively

This blocked the refactoring of the preprocessing, as there was quite some extra logic for considering insertions

## PR Checklist

- [ ] All necessary documentation has been adapted or there is an issue to do so.
- [ ] The implemented feature is covered by an appropriate test.

### CODE REVIEW

**Comment by Taepper** [comment]
I will still add an option to the preprocessingConfig to change the location of the insertions files and make it relative to the input_directory

File: src/silo/preprocessing/sequence_info.cpp
Comment (fengelniederhammer) [inline src/silo/preprocessing/sequence_info.cpp:102]: Is this still a TODO? #220 is already done.

**Review by fengelniederhammer — APPROVED** [review]
Looks good, overall the code got quite a bit simpler :+1: 

Do you also want to adapt the LAPIS docs?
https://lapis.cov-spectrum.org/gisaid/v2/docs/maintainer-docs/references/preprocessing#metadata-file
This still describes the old mechanism.

Also a unit test would be nice that check that the preprocessing works in both paths (also with more than one insertion for a single sequence), but maybe we can leave that to #129?

**Comment by fengelniederhammer** [comment]
And please add `BREAKING CHANGE: ...` in the commit message footer to make sure that we get a new minor version (corresponding to a major version for pre 1.0)

File: src/silo/preprocessing/sequence_info.cpp
Comment (Taepper) [reply src/silo/preprocessing/sequence_info.cpp:102]: I was also under the impression that this already works, but didn't want to remove the TODO without verifying everything first

Comment (fengelniederhammer) [reply src/silo/preprocessing/sequence_info.cpp:102]: Then let's make it a TODO for #129?

Comment (Taepper) [reply src/silo/preprocessing/sequence_info.cpp]: I would add another ticket that request that keys may be left out
But might turn into a won't fix (/ won't implement), as duckdb validation might not be able to cover left out keys in all cases

**Comment by fengelniederhammer** [comment]
Oh and please don't merge yet. We'll have to check whether this is compatible with LAPIS. Most likely, we'll at least have to adapt the preprocessing configs in the deployments.

**Comment by Taepper** [comment]
> Oh and please don't merge yet. We'll have to check whether this is compatible with LAPIS. Most likely, we'll at least have to adapt the preprocessing configs in the deployments.

At least the way it is designed, it should not be breaking for ndjson inputs. The only thing is that we are more strict on the requirement on the json fields of `nucleotideInsertions` and `aminoAcidInsertions`, which should have already been checked before

**Comment by fengelniederhammer** [comment]
Yes, but I just checked, all the configs still contain the `insertions` metadata type, which needs to be removed.

**Comment by fengelniederhammer** [comment]
I squashed and combined the two breaking change lines in the commit message into a single one, because Release Please seems to ignore all but the last of them.

**Pull request merged**
